src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -33,6 +35,72 @@ public static class EnumTests
                 @"""MyUInt64Enum"" : " + ulong.MaxValue +
                 @"}";
 
+        private static readonly string s_jsonByteEnum =
+                @"{" +
+                @"""MyByteEnum"" : " + byte.MaxValue +
+                @"}";
+
+
+        public static IEnumerable<object[]> JsonWithInvalidOutOfBoundNumber
+        {
+            get
+            {
+                const string InvalidNumber = "18446744073709551616"; // ulong.MaxValue + 1;
+                foreach(var i in new dynamic[] { InvalidNumber, ulong.MaxValue })
+                {
+                    yield return new object[] { $"{{ \"MyEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { InvalidNumber })
+                {
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)byte.MaxValue + 2 })
+                {
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)UInt32.MaxValue + 2 })
```

Comment:
nit: It might be cleaner/easier to follow if we had separate tests with strongly typed input values (for the different signed and unsigned numeric types), rather than using dynamic.